src/contracts/core/StakeRootCompendium.sol

```diff
@@ -95,24 +90,30 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
         }
         uint256 numStrategiesAfter = operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length();
 
+        if (isInStakeTree) {
         // make sure they have enough to pay for MIN_PREPAID_PROOFS
-        require(
-            depositInfos[operatorSet.avs][operatorSet.operatorSetId].balance >= minDepositBalance(numStrategiesAfter),
-            "StakeRootCompendium.addOrModifyStrategiesAndMultipliers: insufficient deposit balance"
-        );
-        // note that they've shown increased demand for proofs
-        depositInfos[operatorSet.avs][operatorSet.operatorSetId].lastDemandIncreaseTimestamp = uint32(block.timestamp);
+            require(
+                depositInfos[operatorSet.avs][operatorSet.operatorSetId].balance >= minDepositBalance(numStrategiesAfter),
+                "StakeRootCompendium.addOrModifyStrategiesAndMultipliers: insufficient deposit balance"
+            );
+            // note that they've shown increased demand for proofs
+            depositInfos[operatorSet.avs][operatorSet.operatorSetId].lastDemandIncreaseTimestamp = uint32(block.timestamp);
 
-        // only adding new strategies to count
-        _updateTotalStrategies(numStrategiesBefore, numStrategiesAfter);
+            // only adding new strategies to count
+            _updateTotalStrategies(numStrategiesBefore, numStrategiesAfter);
+        }
     }
 
     /// @inheritdoc IStakeRootCompendium
     function removeStrategiesAndMultipliers(uint32 operatorSetId, IStrategy[] calldata strategies) external {
         // update the deposit balance for the operator set whenever number of strategies is changed
         OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
-        _updateDepositInfo(operatorSet);
+        bool isInStakeTree = _isInStakeTree(operatorSet);
 
+        if (isInStakeTree) {
+            // update the deposit balance for the operator set whenever number of strategies is changed
+            _updateDepositInfo(operatorSet);
```

Comment:
in _updateBalanceInfo, we make sure they're in the stakeTree: https://github.com/Layr-Labs/eigenlayer-contracts/blob/0633e791cf846f363044d706139efb393b7a8c29/src/contracts/core/StakeRootCompendium.sol#L334

Reply:
i prefer not having a pathway where they can edit when they're outside of the stakeTree, it complicates code paths